lib/readyset.rb

```diff
@@ -5,10 +5,91 @@
 require 'readyset/middleware'
 require 'readyset/query'
 require 'readyset/railtie' if defined?(Rails::Railtie)
+require 'readyset/relation_extension'
 
 require 'active_record'
 
 module Readyset
+  attr_writer :configuration
+
+  def self.configuration
+    @configuration ||= Configuration.new
+  end
+
+  def self.configure
+    yield(configuration)
+  end
+
+  # Creates a new cache on ReadySet using the given ReadySet query ID or SQL query. Exactly one of
+  # the `id` or `sql` keyword arguments must be provided.
+  #
+  # This method is a no-op if a cache for the given ID/query already exists.
+  #
+  # @param [String] id the ReadySet query ID of the query from which a cache should be created
+  # @param [String] sql the SQL string from which a cache should be created
+  # @param [String] name the name for the cache being created
+  # @param [Boolean] always whether the cache should always be used. if this is true, queries to
+  # these caches will never fall back to the database
+  # @return [void]
+  # @raise [ArgumentError] raised if exactly one of the `id` or `sql` arguments was not provided
+  def self.create_cache!(id: nil, sql: nil, name: nil, always: false)
+    if (sql.nil? && id.nil?) || (!sql.nil? && !id.nil?)
+      raise ArgumentError, 'Exactly one of the `id` and `sql` parameters must be provided'
+    end
```

Comment:
I agree that helper methods are good for DRYing things up, but IMO, given the small amount of code, given that we are unlikely to use this helper method again anywhere else, and given that the code is self-descriptive since it contains an error message that clearly states its purpose, I'm not sure we gain much by adding a helper method in this case.

If you feel strongly I can definitely add it in -- lmk what you think!